programs/buck_tool.py

```diff
@@ -484,17 +484,17 @@ def _access_check(fn, mode):
 
 _java8 = None
 
-def is_java8():
-    global _java8
-    if _java8 is not None:
-        return _java8
+def is_java8_or_9():
+    global _java8_or_9
+    if _java8_or_9 is not None:
+        return _java8_or_9
     try:
         cmd = ['java', '-Xms64m', '-version']
         output = check_output(cmd, stderr=subprocess.STDOUT)
         version_line = output.strip().splitlines()[0]
-        m = re.compile('(openjdk|java) version "1\.8\..*').match(version_line)
-        _java8 = bool(m)
-        return _java8
+        m = re.compile('(openjdk|java) version "1\.(8|9)\..*').match(version_line)
```

Comment:
I don't expect this to work based on the announced version scheme, and the EA

https://blogs.oracle.com/java-platform-group/entry/a_new_jdk_9_version

```
$ java -version
java version "9-ea"
Java(TM) SE Runtime Environment (build 9-ea+116)
Java HotSpot(TM) 64-Bit Server VM (build 9-ea+116, mixed mode)
```